diff --git a/share/etc/oned.conf b/share/etc/oned.conf
index 55bc020..45d71b9 100644
--- a/share/etc/oned.conf
+++ b/share/etc/oned.conf
@@ -303,10 +303,10 @@ IM_MAD = [
 #    arguments  = "-r 3 -t 15 xen3" ]
 
 # Driver for Xen 4.x
-#IM_MAD = [
-#    name       = "xen",
-#    executable = "one_im_ssh",
-#    arguments  = "-r 3 -t 15 xen4" ]
+IM_MAD = [
+    name       = "xen",
+    executable = "one_im_ssh",
+    arguments  = "-r 3 -t 15 xen4" ]
 
 #-------------------------------------------------------------------------------
 #  XEN SSH-pull Information Driver Manager Configuration
@@ -457,12 +457,12 @@ VM_MAD = [
 #    type       = "xen" ]
 #
 # Driver for Xen 4.x
-#VM_MAD = [
-#    name       = "xen",
-#    executable = "one_vmm_exec",
-#    arguments  = "-t 15 -r 0 xen4",
-#    default    = "vmm_exec/vmm_exec_xen4.conf",
-#    type       = "xen" ]
+VM_MAD = [
+    name       = "xen",
+    executable = "one_vmm_exec",
+    arguments  = "-t 15 -r 0 xen4",
+    default    = "vmm_exec/vmm_exec_xen4.conf",
+    type       = "xen" ]
 #
 #-------------------------------------------------------------------------------
 
